src/generators/utils/parse.ts

```diff
@@ -47,3 +60,100 @@ export function parseSitesFixesConfig<T extends SiteProps>(text: string, options
 
     return sites;
 }
+
+export function indexSitesFixesConfig<T extends SiteProps>(text: string): SitePropsIndex<T> {
+    const domains: {[domain: string]: number | number[]} = {};
+    const domainPatterns: {[domainPattern: string]: number | number[]} = {};
+    const offsets: SitePropsOffsetRecord[] = [];
+
+    function processBlock(recordStart: number, recordEnd: number) {
+        // TODO: more formal definition of URLs and delimiters
+        const block = text.substring(recordStart, recordEnd);
+        const lines = block.split('\n');
+        const commandIndices: number[] = [];
+        lines.forEach((ln, i) => {
+            if (ln.match(/^\s*[A-Z]+(\s[A-Z]+)*\s*$/)) {
+                commandIndices.push(i);
+            }
+        });
+
+        if (commandIndices.length === 0) {
+            return;
+        }
+
+        const urls = parseArray(lines.slice(0, commandIndices[0]).join('\n'));
+        const index = offsets.length;
+        for (const url of urls) {
+            // TODO: make sure this is correct
+            const domain = url.split('/')[0].toLowerCase();
+            if (isFullyQualifiedDomain(domain)) {
+                if (typeof domains[domain] === 'undefined') {
+                    domains[domain] = index;
+                } else if (typeof domains[domain] === 'number') {
+                    domains[domain] = [(domains[domain] as number), index];
+                } else if (typeof domains[domain] === 'object') {
+                    (domains[domain] as number[]).push(index);
+                }
+                continue;
+            }
+
+            if (typeof domains[domain] === 'undefined') {
+                domainPatterns[domain] = index;
+            } else if (typeof domains[domain] === 'number') {
+                domainPatterns[domain] = [(domainPatterns[domain] as number), index];
+            } else if (typeof domains[domain] === 'object') {
+                (domainPatterns[domain] as number[]).push(index);
+            }
+        }
+        offsets.push({
+            start: recordStart,
+            end: recordEnd
+        });
+    }
+
+    let recordStart = 0;
+    // Delimiter between two blocks
+    const delimiterRegex = /\r?\s*={2,}\s*\r?/gm;
+    while (true) {
+        const delimiter = delimiterRegex.exec(text);
```

Comment:
The problem with `while ((delimiter = delimiterRegex.exec(input))) {` is that it skips the very last block because its next delimiter is `null`. I can work around this by explicitly writing another `processBlock()` call.

Reply:
Done!